packages/flutter/lib/src/widgets/text_editing_intents.dart

```diff
@@ -221,6 +221,18 @@ class ExtendSelectionVerticallyToAdjacentPageIntent extends DirectionalCaretMove
   }) : super(forward, collapseSelection);
 }
 
+/// Extends, or moves the current selection from the current
+/// [TextSelection.extent] position to the closest paragraph boundary.
+class ExtendSelectionToParagraphBoundaryIntent extends DirectionalCaretMovementIntent {
+  /// Creates an [ExtendSelectionToParagraphBoundaryIntent].
+  const ExtendSelectionToParagraphBoundaryIntent({
+    required bool forward,
+    required bool collapseSelection,
+    required bool collapseAtReversal,
+    required bool continuesAtWrap,
```

Comment:
Now that I'm looking at these side by side, it bugs me that "collapseAtReversal" and "continuesAtWrap" use different tenses.  It should probably have been "continueAtWrap".  Not the fault of this PR though :)